### PR TITLE
Update WFPSupport.md

### PR DESCRIPTION
**Pull request by wilders-soccerfan**
And add the file WFP_Rule_Editor.png to Media folder

### CODE REVIEW

File: PlusContent/WFPSupport.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+NetworkAccess=Allow;Port=80,443;IP=111.222.333.444 **[Allow any program to access this IP]**
+
+- NetworkAccess=Block;Port=80,443 **[Block rule]** <br>
+NetworkAccess=chrome.exe,Allow;Port=80,443 **[Allow chrome.exe to access any IP]**
+
+- NetworkAccess=Block;Port=80,443 **[Block rule]** <br>
+NetworkAccess=chrome.exe,Allow;Port=80,443;IP=111.222.333.444 **[Allow chrome.exe to access one IP]**
+
+**Block Ports Rule:**
+- NetworkAccess=Block;Port=137,138,139,445 **[This replaces the old BlockPort template]**
```
Comment (isaak654) [inline PlusContent/WFPSupport.md]:
```suggestion
- NetworkAccess=*,Block;Port=137,138,139,445 **[This replaces the old BlockPort template]**
```

As mentioned in [Templates.ini#L3735](https://github.com/sandboxie-plus/Sandboxie/blob/19ab3030da9bee0df075ddf7e30e8135e5673f46/Sandboxie/install/Templates.ini#L3735) (it was updated a number of releases ago).

File: PlusContent/WFPSupport.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+NetworkAccess=Allow:Port=80,443;IP=111.222.333.444 **[Allow any program to access this IP]**
+
+- NetworkAccess=Block;Port=80,443;Protocol=TCP **[Block rule]** <br>
+NetworkAccess=Allow;Port=80,443;IP=111.222.333.444 **[Allow any program to access this IP]**
+
+- NetworkAccess=Block;Port=80,443 **[Block rule]** <br>
+NetworkAccess=chrome.exe,Allow;Port=80,443 **[Allow chrome.exe to access any IP]**
+
+- NetworkAccess=Block;Port=80,443 **[Block rule]** <br>
+NetworkAccess=chrome.exe,Allow;Port=80,443;IP=111.222.333.444 **[Allow chrome.exe to access one IP]**
```
Comment (isaak654) [inline PlusContent/WFPSupport.md]:
```suggestion
- NetworkAccess=*,Block;Port=80,443;IP=0.0.0.0-255.255.255.255 **[No Network Access]** <br> 
NetworkAccess=*,Allow;Port=80,443;IP=111.222.333.444 **[Allow any program to access this IP]**

- NetworkAccess=*,Block;Port=80,443;Protocol=TCP **[Block rule]** <br>
NetworkAccess=*,Allow;Port=80,443;IP=111.222.333.444 **[Allow any program to access this IP]**

- NetworkAccess=*,Block;Port=80,443 **[Block rule]** <br>
NetworkAccess=chrome.exe,Allow;Port=80,443 **[Allow chrome.exe to access any IP]**

- NetworkAccess=*,Block;Port=80,443 **[Block rule]** <br>
NetworkAccess=chrome.exe,Allow;Port=80,443;IP=111.222.333.444 **[Allow chrome.exe to access one IP]**
```

**Review by isaak654 — CHANGES_REQUESTED** [review]

File: PlusContent/WFPSupport.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
+# WFP (Windows Filtering Platform) Support
+
+**Sandboxie Plus v0.9.3** introduced a unique approach to manage network connectivity by implementing 
+not only a **kernel mode** (using a driver) "per box" firewall built on Windows Filtering Platform (WFP) but also a **user mode**, outbound rule-based packet filter. 
+
+### WFP Implementation
+
+To enable WFP functionality, add **NetworkEnableWFP=y** to [Global Settings] of the configuration file
+**sandboxie.ini** and **reboot the machine or reload the driver** for it to take effect. WFP filtering **works for both inbound and outbound** traffic.
```
Comment (isaak654) [inline PlusContent/WFPSupport.md]:
```suggestion
**Sandboxie Plus v0.9.3** introduced a unique approach to manage network connectivity by implementing not only a **kernel mode** (using a driver) "per box" firewall built on Windows Filtering Platform (WFP) but also a **user mode**, outbound rule-based packet filter. 

### WFP Implementation

To enable WFP functionality, add **NetworkEnableWFP=y** to [Global Settings] of the configuration file **Sandboxie.ini** and **reboot the machine or reload the driver** for it to take effect. WFP filtering **works for both inbound and outbound** traffic.
```

File: PlusContent/WFPSupport.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+- To exempt blocking for a box, add **AllowNetworkAccess=y** to [Box Settings]
+- To allow a selected program in a box, add **AllowNetworkAccess=program.exe,y** to [Box Settings]
+- To block a selected program in a box, add **AllowNetworkAccess=program.exe,n** to [Box Settings]
+
+**Limitations of the WFP implementation:** WFP will filter **only** TCP/UDP protocols. The WFP filter rules can be implemented by restricting communication only to **specified IP addresses or selected ports** by using a rule based hierarchy based on **"NetworkAccess=..."** (as described later). Restricted boxed processes will still be able to resolve domain names using the system service but will not be able to send or receive data packets directly.
+
+### User Mode Packet Filter Implementation
+
+**Sandboxie Plus v0.9.3** also added a fully functional rule-based packet filter in user mode for the 
+case when **NetworkEnableWFP=y** is not set. This mechanism also replaces the primitive **"BlockPort=..."** functionality of older versions.
```
Comment (isaak654) [inline PlusContent/WFPSupport.md]:
```suggestion
**Sandboxie Plus v0.9.3** also added a fully functional rule-based packet filter in user mode for the case when **NetworkEnableWFP=y** is not set. This mechanism also replaces the primitive **"BlockPort=..."** functionality of older versions.
```

File: PlusContent/WFPSupport.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+**Caveat:** For reliable isolation, **the use of kernel mode WFP-based filtering is strongly recommended**.
+
+**The Rationale for Two Filtering Modes:** 
+The rationale for implementing network functionality in both user mode and kernel mode (driver) 
+is twofold. First, it allows for easier debugging of the rule processing code 
+(simpler to debug in user mode) as both modes use the same code to make decisions based on the 
+preset rules. 
+Second, the WFP callouts are global i.e. they are triggered for any process on the system whether 
+sandboxed or not. In the latter case they don't do anything and the use of a hash map to 
+identify sandboxed programs that require action can provide optimal performance.
```
Comment (isaak654) [inline PlusContent/WFPSupport.md]:
```suggestion
The rationale for implementing network functionality in both user mode and kernel mode (driver) is twofold. First, it allows for easier debugging of the rule processing code (simpler to debug in user mode) as both modes use the same code to make decisions based on the preset rules. 
Second, the WFP callouts are global i.e. they are triggered for any process on the system whether sandboxed or not. In the latter case they don't do anything and the use of a hash map to identify sandboxed programs that require action can provide optimal performance.
```

File: PlusContent/WFPSupport.md [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+sandboxed or not. In the latter case they don't do anything and the use of a hash map to 
+identify sandboxed programs that require action can provide optimal performance. 
+Users of a 3rd party firewall may not feel comfortable turning their firewall off and they can continue to use it and still benefit from the selective "per sandbox" network rules offered by Sandboxie Plus. For example, Box1 may allow network access for Program1, while in Box2 the same Program1 may be blocked or even allowed but with a different set of rules for network access.
+
+**Combining WFP with User Mode Filtering:** 
+Does enabling WFP support do anything without user set IP/Port based rules?
+Indeed it does! <br>
+If you set "block internet access" for a given process and have the driver (for WFP) enabled, 
+you can select for that box which method to apply: using WFP or blocking network devices. Even though the approach of blocking the network device end points is more absolute, it has been known 
+to cause some applications to crash.
```
Comment (isaak654) [inline PlusContent/WFPSupport.md]:
```suggestion
**Combining WFP with User Mode Filtering:** 
Does enabling WFP support do anything without user set IP/Port based rules? Indeed it does! <br>
If you set "block internet access" for a given process and have the driver (for WFP) enabled, you can select for that box which method to apply: using WFP or blocking network devices. Even though the approach of blocking the network device endpoints is more absolute, it has been known to cause some applications to crash.
```

File: PlusContent/WFPSupport.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+### WFP and Multiple Firewalls
+
+Windows Filtering Platform (WFP) is configured by installing a **provider** of filter rules.
+WFP based firewalls may use the standard Windows Firewall's provider or create their own.
+Some firewalls use WFP at the user mode level (no drivers), while others use WFP in kernel mode 
+(based on their own driver). If several kernel mode firewalls are installed and active at the same time, each driver installs its own callout functions at the positions in the network stack it wants to control and all those functions are then called by the kernel for ALL the drivers (providers). This results in an amalgamation of rules set by each firewall. An inbuilt arbitration mechanism in WFP then decides which rules take precedence.
```
Comment (isaak654) [inline PlusContent/WFPSupport.md]:
```suggestion
Windows Filtering Platform (WFP) is configured by installing a **provider** of filter rules. <br>
WFP based firewalls may use the standard Windows Firewall's provider or create their own. <br>
Some firewalls use WFP at the user mode level (no drivers), while others use WFP in kernel mode (based on their own driver). If several kernel mode firewalls are installed and active at the same time, each driver installs its own callout functions at the positions in the network stack it wants to control and all those functions are then called by the kernel for ALL the drivers (providers). This results in an amalgamation of rules set by each firewall. An inbuilt arbitration mechanism in WFP then decides which rules take precedence.
```

File: PlusContent/WFPSupport.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+The **attributes** at our disposal (with some examples of syntax) are:
+- **Action**=Allow|Block [Selected from the Network Restrictions tab]
+- **Program**=program.exe
+- **Port**=80,443,1000-2000
+- **IP**=111.222.333.444,0.0.0.0-255.255.255.255
+- **Protocol**=TCP|UDP 
+
+The following **rules precedence** scheme determines rule hierarchy:
+1. A rule for a specified program trumps a rule for all programs except a given one, 
+   trumps rules for all programs.
```
Comment (isaak654) [inline PlusContent/WFPSupport.md]:
```suggestion
1. A rule for a specified program trumps a rule for all programs except a given one, trumps rules for all programs.
```

File: PlusContent/WFPSupport.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+2. A rule with a Port or IP trumps a rule without.
```
Comment (isaak654) [inline PlusContent/WFPSupport.md]:
```suggestion
2. A rule with a Port or IP trumps a rule without:
```

File: PlusContent/WFPSupport.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+- **IP**=111.222.333.444,0.0.0.0-255.255.255.255
+- **Protocol**=TCP|UDP 
+
+The following **rules precedence** scheme determines rule hierarchy:
+1. A rule for a specified program trumps a rule for all programs except a given one, 
+   trumps rules for all programs.
+2. A rule with a Port or IP trumps a rule without.  
+    2a. A rule with IP and Port trumps a rule with IP only or Port only.  
+    2b. A rule with one IP trumps a rule with an IP range that is besides that on 
+        the same level.
```
Comment (isaak654) [inline PlusContent/WFPSupport.md]:
```suggestion
    2b. A rule with one IP trumps a rule with an IP range that is besides that on the same level.
```

**Review by isaak654 — CHANGES_REQUESTED** [review]

File: PlusContent/WFPSupport.md (hunk 9 above)
Comment (isaak654) [reply PlusContent/WFPSupport.md]: I am not sure what the sentence means, so I may be wrong.

File: PlusContent/WFPSupport.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
+# WFP (Windows Filtering Platform) Support
+
+**Sandboxie Plus v0.9.3** introduced a unique approach to manage network connectivity by implementing 
+not only a **kernel mode** (using a driver) "per box" firewall built on Windows Filtering Platform (WFP) but also a **user mode**, outbound rule-based packet filter.
```
Comment (isaak654) [inline PlusContent/WFPSupport.md:4]:
```suggestion
**Sandboxie Plus v0.9.3** introduced a unique approach to manage network connectivity by implementing not only a **kernel mode** (using a driver) "per box" firewall built on Windows Filtering Platform (WFP) but also a **user mode**, outbound rule-based packet filter.
```

**Review by isaak654 — CHANGES_REQUESTED** [review]

File: PlusContent/WFPSupport.md (hunk 3 above)
Comment (isaak654) [reply PlusContent/WFPSupport.md:9]: Bump

**Review by isaak654 — CHANGES_REQUESTED** [review]
Previously I noticed an excess of white spaces, so there may be more. Please open the .md file with Notepad++, go to the Macro menu, select `Trim Trailing Space and Save` and reupload it.

**Comment by wilders-soccerfan** [comment]
Updated WFPSupport.md to put use of [DefaultBox] in context in lines 7-13 (if you approve).
[WFPSupport.md](https://github.com/sandboxie-plus/sandboxie-docs/files/11791372/WFPSupport.md)

File: PlusContent/WFPSupport.md [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+
+**Combining WFP with User Mode Filtering:**
+If you set "block internet access" for a given process and have the driver (for WFP) enabled, you can select for that box which method to apply: using WFP or blocking network devices. Even though the approach of blocking the network device endpoints is more absolute, it has been known to cause some applications to crash.
+
+### WFP and Multiple Firewalls
+
+Windows Filtering Platform (WFP) is configured by installing a **provider** of filter rules. <br>
+WFP based firewalls may use the standard Windows Firewall's provider or create their own. <br>
+Some firewalls use WFP at the user mode level (no drivers), while others use WFP in kernel mode (based on their own driver). If several kernel mode firewalls are installed and active at the same time, each driver installs its own callout functions at the positions in the network stack it wants to control and all those functions are then called by the kernel for ALL the drivers (providers). This results in an amalgamation of rules set by each firewall. An inbuilt arbitration mechanism in WFP then decides which rules take precedence.
```
Comment (isaak654) [inline PlusContent/WFPSupport.md]: About lines 36-37-38, could you find a way to better integrate them into the paragraph as a whole? They seem too distant to be part of the same discourse (with or without the br tags).

I wouldn't know how to do it, but I think it should be done.

**Review by isaak654 — CHANGES_REQUESTED** [review]

File: PlusContent/WFPSupport.md (hunk 12 above)
Comment (wilders-soccerfan) [reply PlusContent/WFPSupport.md]: I see your point! Does the attached file (with your advice and some reshuffling of lines 29-40) make a little more sense about WFP firewalls, how they work  (this is not generally well explained) and why Sandboxie can be used simultaneously with one's favorite firewall?
Edit: Also added punctuation marks (periods) to lines 9-13.
[WFPSupport.md](https://github.com/sandboxie-plus/sandboxie-docs/files/11843440/WFPSupport.md)

**Review by isaak654 — APPROVED** [review]

**Pull request merged**